### PR TITLE
[MRG] improve DimensionError exception message

### LINKED ISSUE #1063: concat_imgs throws incomprehensible error for incongruent nifti shapes
I just realized that concat_imgs wants the nifti items to concat to exhibit a same dimension. The code below tried to concatenate ``all_frames`` (4D) and ``block_mean_frame`` (3D), yet failed. Perhaps we could improve the error message at this point. What do you guys think?

```python
In [158]: all_frames = concat_imgs([all_frames, block_mean_frame])
---------------------------------------------------------------------------
DimensionError                            Traceback (most recent call last)
<ipython-input-158-9e4d213099ce> in <module>()
----> 1 all_frames = concat_imgs([all_frames, block_mean_frame])

/home/parietal/dbzdok/.local/lib/python2.7/site-packages/nilearn-0.1.5.dev0-py2.7.egg/nilearn/_utils/niimg_conversions.pyc in concat_niimgs(niimgs, dtype, ensure_ndim, memory, memory_level, auto_resample, verbose)
    410         # We check the dimensionality of the niimg
    411         try:
--> 412             niimg = check_niimg(niimg, ensure_ndim=ndim)
    413         except DimensionError as exc:
    414             # Keep track of the additional dimension in the error

/home/parietal/dbzdok/.local/lib/python2.7/site-packages/nilearn-0.1.5.dev0-py2.7.egg/nilearn/_utils/niimg_conversions.pyc in check_niimg(niimg, ensure_ndim, atleast_4d, dtype, return_iterator, wildcards)
    253 
    254     if ensure_ndim is not None and len(niimg.shape) != ensure_ndim:
--> 255         raise DimensionError(len(niimg.shape), ensure_ndim)
    256 
    257     if return_iterator:

DimensionError: Data must be a 5D Niimg-like object but you provided a list of 3D images. See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.```

### PR DESCRIPTION
**Pull request by aabadie**
Should fix #1063

### CODE REVIEW

File: nilearn/_utils/exceptions.py [hunk 1]
```diff
@@ -49,14 +49,16 @@ def increment_stack_counter(self):
 
     @property
     def message(self):
-        return ("Data must be a {0}D Niimg-like object but you provided a "
-                "{1}{2}D image{3}. "
+        return ("Input data has incompatible dimensionality: "
+                "Expected dimention is {0}D and you provided a "
```
Comment (AlexandreAbraham) [inline nilearn/_utils/exceptions.py]: dimension

Comment (aabadie) [inline nilearn/_utils/exceptions.py]: oh nooo ! (the issue is repeated in so many places)

Comment (aabadie) [inline nilearn/_utils/exceptions.py]: I pushed an update.

**Comment by banilo** [comment]
LGTM

**Comment by aabadie** [comment]
I found I had this one pending for a moment. Once CI are green, could someone have a look ?  (@AlexandreAbraham, @banilo)

**Comment by AlexandreAbraham** [comment]
Good to go, thanks!

**Pull request merged**